plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,403 @@
+        - This is only required when creating a new Port Channel Interface.
+        type: str
+        aliases: [ policy_uuid, pc_policy_uuid ]
+    interface_descriptions:
+        description:
+        - The list of descriptions for each interface.
+        - The interface must exist when a description is provided.
+        type: list
+        elements: dict
+        suboptions:
+            interface_id:
+                description:
+                - The interface ID.
+                type: str
+            description:
+                description:
+                - The description of the interface.
+                type: str
+    state:
+        description:
+        - Use C(absent) for removing.
+        - Use C(query) for listing an object or multiple objects.
+        - Use C(present) for creating or updating.
+        type: str
+        choices: [ absent, query, present ]
+        default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    description: My Ansible Port Channel
+    port_channel_interface: ansible_port_channel_interface
+    node: 101
+    interfaces:
+      - 1/1
+      - 1/10-11
+    interface_policy_uuid: ansible_port_channel_policy
+    interface_descriptions:
+      - interface_id: 1/1
+        description: My first Ansible Interface
+      - interface_id: 1/10
+        description: My second Ansible Interface
+      - interface_id: 1/11
+        description: My third Ansible Interface
+    state: present
+
+- name: Query an Port Channel Interface with template_name
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: query
+  register: query_one
+
+- name: Query all Port Channel Interfaces in the template
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete an Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def lookup_valid_interfaces(interfaces):
```

Comment:
This can be moved to mso.py since the same function will be used by another module

Reply:
I will move it to mso.py.